### PR TITLE
Create bug triaging process

### LINKED ISSUE #2275: Establish bug prioritisation process
### Description
Setup a bug prioritisation/triage process and apply to existing bugs.

https://wiki.hyperledger.org/display/BESU/Defect+Prioritisation+Policy is likely to be a good starting point.

### PR DESCRIPTION
**Pull request by cemozerr**
## PR Description
Create markdown document that outlines Teku bug triaging.

## Fixed Issue(s)
Fixes https://github.com/PegaSysEng/teku/issues/2275

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: BugTriaging.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Teku Bug Triage Process
+
+In Teku, we have a well-defined policy regarding bug categorization. In a nutshell, our bugs are categorized into 5 priorities. 
+
+#### Bug Priorities
+
+| Priority | Examples |
+|-|-|
+| Very High (P1) | <ul><li>Affects consensus</li> <li>Breaks sync</li><li>Security issue</li></ul> |
```
Comment (ajsutton) [inline BugTriaging.md:171]: Probably want to keep:
Non-planned backward incompatibilities, preventing the node from participating in the network

File: BugTriaging.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+| High (P2) | <ul><li> Degrades validator rewards</li> <li>Unexpected behaviour of core features</li></ul>|
```
Comment (ajsutton) [inline BugTriaging.md]: I'd add "or incurs penalties" to the validator rewards point and add "Significant performance regression" or maybe just "Performance degradation that affects ability to participate in the network"

**Review by ajsutton — COMMENTED** [review]
I think we wound up linking to the wrong Besu page.  The key piece we want is the policy which is a more abstract definition of how things are prioritised - this page just has examples of things that would be in each category.

https://wiki.hyperledger.org/display/BESU/Defect+Prioritisation+Policy

File: BugTriaging.md (hunk 1 above)
Comment (cemozerr) [reply BugTriaging.md:171]: Done.

File: BugTriaging.md (hunk 2 above)
Comment (cemozerr) [reply BugTriaging.md]: Done.

**Comment by cemozerr** [comment]
> I think we wound up linking to the wrong Besu page.

As discussed offline, this was on purpose, but I've also added the abstract concepts to fall back on to aid categorization process.

File: BugTriaging.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+#### Deciding Priority
+When deciding how to respond to a bug, consider the two variables of probability and severity, and apply them to a risk matrix to determine the priority and corresponding action.
+
+#### Probability
+How often actual customers are likely to encounter the issue during their use of the product?
+
+A feature rarely used, with a high likelihood of failure, may have a lower overall probability than a popular feature with a low likelihood of failure (due to a high number of customers affected).
+
+| Frequent   | Will occur several times in the lifetime of the product.          |
```
Comment (ajsutton) [inline BugTriaging.md]: The formatting on this table isn't quite right - the first line is being used as a header.  You can embed most HTML in markdown which may be the easiest way to get the tables to format correctly.

File: BugTriaging.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+| Remote     | Unlikely but possible to occur in the lifetime of the product.    |
+| Improbable | So unlikely, it can be assumed occurrence may not be experienced. |
+
+#### Severity
+
+How bad is the problem when it is encountered?
+
+An issue that causes data loss/corruption is automatically classed as Catastrophic.
+
+| Catastrophic  | Detrimental risk for the product as a whole.                             |
```
Comment (ajsutton) [inline BugTriaging.md]: Same issue with this table.

**Review by ajsutton — APPROVED** [review]
LGTM.  Just need to work out how to make the table format work.

File: BugTriaging.md (hunk 3 above)
Comment (cemozerr) [reply BugTriaging.md]: Good catch. Will fix.

**Pull request merged**